### DISCUSSION TITLE
Is there a plan about adding macro programming courses?

### DISCUSSION THREAD

**Discussion by JCKeep** [google/comprehensive-rust#888] [Ideas]
Macro programming is one of the hardest skills to master in Rust, and I hope there are courses to get beginners up to speed faster.

**Comment by mgeisler** [comment]
Hi @JCKeep, there is no concrete plan right now, bit macros is something I would love to see covered.

We have #32 where many different ideas were discussed. I would be happy to see someone contribute a chapter about macros. This could fill a morning slot, or it could be an hour, completely up to the person.

Additional topics like that could be taught on demand at Google and elsewhere.
